### PR TITLE
Search for products#181

### LINKED ISSUE #181: Search By Product Name
Users should be able to search by product name

### PR DESCRIPTION
**Pull request by Abazhenov**
closes #181 
basic search functionality implemented, objects are console logged if they exist in the database

### CODE REVIEW

**Comment by Abazhenov** [comment]
Search bar now takes search, finds squares, and redirects to a state where we see the search results, might need to refactor the two states into one but for now this works.

File: browser/js/squares/squares.js [hunk 1]
```diff
@@ -22,7 +35,16 @@ app.factory('SquaresFactory', function($http) {
         squares: function() {
             return $http.get('/api/squares').then((tags) => tags.data)
                 // Add error catcher
-                .catch();
+                .catch(function (e){
+                  if (e) console.log(e)
+                });
+        },
+        search: name => {
+          return $http.get(`/api/squares/search/${name}`).then(squares => squares.data)
```
Comment (connieelee) [inline browser/js/squares/squares.js]: probably "res" is better than "squares" considering what comes back in that promise

File: browser/js/squares/squares.js [hunk 2]
```diff
@@ -22,7 +35,16 @@ app.factory('SquaresFactory', function($http) {
...
+          .catch(function(e){
```
Comment (connieelee) [inline browser/js/squares/squares.js]: you can also just inject $log and catch($log.error)

Comment (connieelee) [inline browser/js/squares/squares.js]: (just a suggestion)

**Comment by connieelee** [comment]
comments you can consider but just gonna merge now.

**Pull request merged**